editor.planx.uk/src/lib/graphql.ts

```diff
@@ -87,3 +88,13 @@ export const client = new ApolloClient({
   link: from([retryLink, errorLink, authLink, httpLink]),
   cache: new InMemoryCache(),
 });
+
+/**
+ * Explicitly connect to Hasura using the "public" role
+ * Allows authenticated users with a different x-hasura-default-role (e.g. teamEditor, platformAdmin) to access public resources
+ */
+export const publicContext: DefaultContext = {
```

Comment:
What's going on here?

In order to keep things a little simpler, the newly created roles do not have the same set of permissions as the public role. This would mean any change to the public role (e.g. new tables and columns) would need to get propagated across all other roles. Maintenance would be hard and not much fun! I had hoped by just adding the `public` role to a user's `x-hasura-allowed-role` it would automatically fall back to this, but there's no concept of role "hierarchy" like this.

In a perfect would it would be nice to user [inherited roles](https://hasura.io/docs/latest/auth/authorization/inherited-roles/) (meaning the actual `teamAdmin` role could be made of `public` and an abstract `_teamAdmin`. I tried getting this to work but changes made in Hasura weren't persisting and updating any metadata files. We might need to [update Hasura to the metadata v3 config](https://hasura.io/docs/latest/migrations-metadata-seeds/legacy-configs/upgrade-v3/) for this to work.

I'm happy with this solution for now - we always know what's a public query.

Reply:
I think the approach in https://github.com/theopensystemslab/planx-new/pull/2195 of just using an `authLink` and 
`publicLink` is a little simpler than modifying the context.

Once the other PR is merged I'll rebase this to adopt that approach.